common/types/chain.go

```diff
@@ -4,9 +4,10 @@ import "fmt"
 
 // Sequence represents the base type, for any chain's sequence object.
 // It should be convertible to a string
-type Sequence interface {
+type Sequence[SEQ any] interface {
```

Comment:
No need to add generic type here.
type Sequence interface {
    ...
    Next() Sequence
}

Reply:
It is required in order to support a Next method that returns a self referential type. This is the main reason that I don't support using a method for nonce generation.

Reply:
Hmm, I didn't understand the blocker.
Can't we use the same pattern used in Head interface, for this method:
https://github.com/smartcontractkit/chainlink/blob/b3d385b34c5269998531f219567d07722544f925/common/types/head.go#L17-L18

Reply:
We are and it creates a huge mess by leaking the generic param everywhere the nonce is used. There is no need to use this pattern for this case, and I don't see the benefits.